File: apps/mobile/lib/domain/entities/transaction.dart
enum TransactionType {
  expense,
  income,
  transfer,
}

class Transaction {
  final int id;
  final int? groupId;
  final int ownerUserId;
  final TransactionType type;
  final DateTime date;
  final int amount;
  final String? currencyCode;
  final int? originalAmount;
  final int? categoryId;
  final int? tagId;
  final int? recurringRuleId;
  final int? receiptId;
  final String? merchant;
  final String? memo;
  final DateTime createdAt;
  final DateTime updatedAt;

  Transaction({
    required this.id,
    this.groupId,
    required this.ownerUserId,
    required this.type,
    required this.date,
    required this.amount,
    this.currencyCode,
    this.originalAmount,
    this.categoryId,
    this.tagId,
    this.recurringRuleId,
    this.receiptId,
    this.merchant,
    this.memo,
    required this.createdAt,
    required this.updatedAt,
  });

  Transaction copyWith({
    int? id,
    int? groupId,
    int? ownerUserId,
    TransactionType? type,
    DateTime? date,
    int? amount,
    String? currencyCode,
    int? originalAmount,
    int? categoryId,
    int? tagId,
    int? recurringRuleId,
    int? receiptId,
    String? merchant,
    String? memo,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Transaction(
      id: id ?? this.id,
      groupId: groupId ?? this.groupId,
      ownerUserId: ownerUserId ?? this.ownerUserId,
      type: type ?? this.type,
      date: date ?? this.date,
      amount: amount ?? this.amount,
      currencyCode: currencyCode ?? this.currencyCode,
      originalAmount: originalAmount ?? this.originalAmount,
      categoryId: categoryId ?? this.categoryId,
      tagId: tagId ?? this.tagId,
      recurringRuleId: recurringRuleId ?? this.recurringRuleId,
      receiptId: receiptId ?? this.receiptId,
      merchant: merchant ?? this.merchant,
      memo: memo ?? this.memo,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: apps/api/src/domain/models/transaction.rs
use chrono::{NaiveDate, NaiveDateTime};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Copy, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "enum", rename_all = "UPPERCASE")]
pub enum TransactionType {
    Expense,
    Income,
    Transfer,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Transaction {
    pub id: i64,
    pub group_id: Option<i64>,
    pub owner_user_id: i64,
    #[sqlx(rename = "type")]
    pub transaction_type: TransactionType,
    pub date: NaiveDate,
    pub amount: i64,
    pub currency_code: Option<String>,
    pub original_amount: Option<i64>,
    pub category_id: Option<i64>,
    pub tag_id: Option<i64>,
    pub recurring_rule_id: Option<i64>,
    pub receipt_id: Option<i64>,
    pub merchant: Option<String>,
    pub memo: Option<String>,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

